### PR TITLE
[Test] Fix Docker scripts crashing when variables not set

### LINKED ISSUE #5515: [Testing] Strict bash flags cause docker scripts to fail locally
`test/dockerized-unit-tests.sh` checks if the `$DEBUG` variable is given a certain value to determine how it should run.

This doesn't work after #5051  because the script will fail when this variable is not set.

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

A recent PR causes strict checking of variables in bash. We are doing some sneaky environment variable tricks relying on variables sometimes being set so this made our scripts crash.

They should now work as before.

#### Testing instructions (if applicable)

1. Running `npm run tests:unit` right now should cause an 'unbound variable error'
2. You should be able to run `npm run tests:unit` locally without that error (though the tests may fail for other reasons)

#### Links to related tickets (GitHub, Redmine, ...)

* Resolves #5515

### CODE REVIEW

File: test/dockerized-integration-tests.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-if [ "$DEBUG" == "true" ]
+if [ -v DEBUG ];
```
Comment (driusan) [inline test/dockerized-integration-tests.sh]: What does this do? I can't find any documentation of "[" or "test" having  "-v" option anywhere.

Comment (johnsaigle) [reply test/dockerized-integration-tests.sh]: I just found it from searching how to check whether a variable is set in bash. Here's an example from Stack Exchange

https://unix.stackexchange.com/a/212192

Comment (johnsaigle) [reply test/dockerized-integration-tests.sh]: I tested it manually and the scripts work for me now where before they would crash.

File: Dockerfile.test.php7.debug [hunk 2]
```diff
@@ -1,9 +1,9 @@
-FROM php:7.0
+FROM php:7.2
```
Comment (johnsaigle) [inline Dockerfile.test.php7.debug]: Running this file (`npm run tests:unit:debug`) fails without these changes.

Comment (driusan) [reply Dockerfile.test.php7.debug]: This made me nervous until I checked the .travis.yml and you're right, it's doing

`- sed -i "s/7.2/$TRAVIS_PHP_VERSION/g" Dockerfile.test.php7` (7.2, not 7.0).. so how is this working now?

**Pull request merged**